xsnippet/api/services/snippet.py

```diff
@@ -47,23 +47,41 @@ def __init__(self, db):
         self.db = db
 
     async def create(self, snippet):
-        snippet = self._normalize(snippet)
-
         now = datetime.datetime.utcnow().replace(microsecond=0)
+
+        snippet = self._normalize(snippet)
+        snippet['changesets'] = [
+            {
+                'content': snippet.pop('content'),
+                'created_at': now,
+            },
+        ]
         snippet['created_at'] = now
         snippet['updated_at'] = now
 
         snippet_id = await self.db.snippets.insert(snippet)
         snippet['id'] = snippet_id
+        snippet['content'] = snippet.pop('changesets', [])[0]['content']
+
         return snippet
 
     async def update(self, snippet):
         now = datetime.datetime.utcnow().replace(microsecond=0)
         snippet['updated_at'] = now
 
+        condition = {'$set': snippet}
```

Comment:
nit: it's not a condition, it's the `update` parameter - a document that describes what updates must be performed

Reply:
Will rename